modules/dataframe/src/util/dataframeutil.cpp

```diff
@@ -200,8 +203,8 @@ std::vector<std::vector<std::uint32_t>> getMatchingRows(std::shared_ptr<const Co
         auto catCol2 = dynamic_cast<const CategoricalColumn*>(rightCol.get());
         IVW_ASSERT(catCol2, "right column is not categorical");
 
-        auto valuesRight = catCol2->getValues();
-        for (auto&& [i, key] : util::enumerate(catCol1->getValues())) {
+        auto valuesRight = catCol2->values();
+        for (auto&& [i, key] : util::enumerate(catCol1->values())) {
```

Comment:
this is an O(n^2) implementation, should be possible to to in nlogn at least... are we using this on any non trivial datasets?